### PR TITLE
HTTPCORE-670 -  Add new headers

### PR DESCRIPTION
**Pull request by arturobernalg**
Add new header

Set-Cookie --> https://tools.ietf.org/html/rfc2109#section-4.2.2
Link --> href="https://tools.ietf.org/html/rfc5988
Cookie --> https://tools.ietf.org/html/rfc2109
Content-Disposition --> https://tools.ietf.org/html/rfc6266
Access-Control-Request-Method --> https://www.w3.org/TR/cors
Access-Control-Request-Headers  --> https://www.w3.org/TR/cors
Access-Control-Max-Age --> https://www.w3.org/TR/cors
Access-Control-Expose-Headers --> https://www.w3.org/TR/cors
Access-Control-Allow-Origin --> https://www.w3.org/TR/cors
Access-Control-Allow-Methods --> https://www.w3.org/TR/cors
Access-Control-Allow-Credentials -->https://www.w3.org/TR/cors

### CODE REVIEW

File: httpcore5/src/main/java/org/apache/hc/core5/http/HttpHeaders.java [hunk 1]
```diff
@@ -130,6 +188,17 @@ private HttpHeaders() {
 
     public static final String STATUS_URI = "Status-URI";
 
+    /**
+     * The HTTP {@code Set-Cookie} header field name.
+     * @see <a href="https://tools.ietf.org/html/rfc2109#section-4.2.2">Section 4.2.2 of RFC 2109</a>
```
Comment (ok2c) [inline httpcore5/src/main/java/org/apache/hc/core5/http/HttpHeaders.java:193]: @arturobernalg I am not sure about references to RFCs in the source code. For instance, HttpClient conforms to rfc6265 and not rfc2109. I would not want to scan the source code for RFC references every time protocol level gets upgraded.

File: httpcore5/src/main/java/org/apache/hc/core5/http/HttpHeaders.java [hunk 2]
```diff
@@ -130,6 +188,17 @@ private HttpHeaders() {
...
+     */
+    public static final String SET_COOKIE = "Set-Cookie";
+    /**
+     * The HTTP {@code Set-Cookie2} header field name.
+     * @see <a href="https://tools.ietf.org/html/rfc2965">RFC 2965</a>
+     */
+    public static final String SET_COOKIE2 = "Set-Cookie2";
```
Comment (ok2c) [inline httpcore5/src/main/java/org/apache/hc/core5/http/HttpHeaders.java:200]: @arturobernalg `Set-Cookie2` is considered obsolete and should not be used. I would not include it,

**Review by ok2c — CHANGES_REQUESTED** [review]

File: httpcore5/src/main/java/org/apache/hc/core5/http/HttpHeaders.java (hunk 2 above)
Comment (arturobernalg) [reply httpcore5/src/main/java/org/apache/hc/core5/http/HttpHeaders.java]: Agree, changed

File: httpcore5/src/main/java/org/apache/hc/core5/http/HttpHeaders.java (hunk 1 above)
Comment (arturobernalg) [reply httpcore5/src/main/java/org/apache/hc/core5/http/HttpHeaders.java]: Agree, changed

**Review by ok2c — APPROVED** [review]

**Pull request merged**